### PR TITLE
Avoid cloning element for optionText

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx [hunk 1]
```diff
@@ -496,16 +498,9 @@ If you provided a React element for the optionText prop, you must also provide t
...
-                    if (isValidElement(optionText)) {
-                        return cloneElement(optionText, {
-                            record: record as RaRecord,
-                            ...props,
-                        });
-                    }
-
-                    return <li {...props}>{getChoiceText(record)}</li>;
-                }}
+                renderOption={(props, record) => (
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:501]: You are not asking for `optionText` to return it

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx [hunk 2]
```diff
@@ -60,8 +60,8 @@ import { LinearProgress } from '../layout';
  * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
  * <CheckboxGroupInput source="recipients" choices={choices} optionText={optionRenderer} />
  *
- * `optionText` also accepts a React Element, that will be cloned and receive
- * the related choice as the `record` prop. You can use Field components there.
+ * `optionText` also accepts a React Element, that can access the
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx:63]:
```suggestion
 * `optionText` also accepts a React Element, that can access 
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/field/SelectField.tsx [hunk 3]
```diff
@@ -43,8 +43,8 @@ import { PublicFieldProps, InjectedFieldProps, fieldPropTypes } from './types';
  * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
  * <SelectField source="author_id" choices={choices} optionText={optionRenderer} />
  *
- * `optionText` also accepts a React Element, that will be cloned and receive
- * the related choice as the `record` prop. You can use Field components there.
+ * `optionText` also accepts a React Element, that can access the
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/field/SelectField.tsx:46]:
```suggestion
 * `optionText` also accepts a React Element, that can access 
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.tsx [hunk 4]
```diff
@@ -58,15 +58,15 @@ import { LinearProgress } from '../layout';
  * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
  * <CheckboxGroupInput source="recipients" choices={choices} optionText={optionRenderer} />
  *
- * `optionText` also accepts a React Element, that will be cloned and receive
- * the related choice as the `record` prop. You can use Field components there.
+ * `optionText` also accepts a React Element, that can access the
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/RadioButtonGroupInput.tsx:61]:
```suggestion
 * `optionText` also accepts a React Element, that can access 
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/SelectArrayInput.tsx [hunk 5]
```diff
@@ -62,8 +62,8 @@ import {
  * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
  * <SelectArrayInput source="authors" choices={choices} optionText={optionRenderer} />
  *
- * `optionText` also accepts a React Element, that will be cloned and receive
- * the related choice as the `record` prop. You can use Field components there.
+ * `optionText` also accepts a React Element, that can access the
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.tsx:65]:
```suggestion
 * `optionText` also accepts a React Element, that can access 
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx (hunk 1 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:501]: I don't understand your question

File: packages/ra-ui-materialui/src/input/SelectInput.tsx [hunk 6]
```diff
@@ -60,15 +60,18 @@ import {
  * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
  * <SelectInput source="author_id" choices={choices} optionText={optionRenderer} />
  *
- * `optionText` also accepts a React Element, that will be cloned and receive
- * the related choice as the `record` prop. You can use Field components there.
+ * `optionText` also accepts a React Element, that can access the
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/SelectInput.tsx:63]:
```suggestion
 * `optionText` also accepts a React Element, that can access 
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx (hunk 1 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:501]: That's what `getChoiceText` does

Comment (WiXSL) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:501]: Dah, I of course, my bad!

**Review by WiXSL — APPROVED** [review]

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx [hunk 7]
```diff
@@ -496,16 +498,9 @@ If you provided a React element for the optionText prop, you must also provide t
                 onChange={handleAutocompleteChange}
                 onBlur={field.onBlur}
                 onInputChange={handleInputChange}
-                renderOption={(props, record) => {
-                    if (isValidElement(optionText)) {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:500]: Why did you remove this one? This removes the ability to pass an element as optionText

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx (hunk 7 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:500]: Nope, see `getChoiceText`

**Comment by fzaninotto** [comment]
Needs rebase
